index.bs

```diff
@@ -2297,6 +2312,174 @@ create them does not matter.
  1. Return « |branch1|, |branch2| ».
 </div>
 
+<div algorithm>
+ <dfn abstract-op lt="ReadableByteStreamTee">ReadableByteStreamTee(|stream|)</dfn>
+ performs the following steps:
+
+ 1. Assert: |stream| [=implements=] {{ReadableStream}}.
+ 1. Assert: |stream|.[=ReadableStream/[[controller]]=] [=implements=]
+    {{ReadableByteStreamController}}.
+ 1. Let |reader| be ? [$AcquireReadableStreamDefaultReader$](|stream|).
+ 1. Let |reading| be false.
+ 1. Let |canceled1| be false.
+ 1. Let |canceled2| be false.
+ 1. Let |reason1| be undefined.
+ 1. Let |reason2| be undefined.
+ 1. Let |branch1| be undefined.
+ 1. Let |branch2| be undefined.
+ 1. Let |cancelPromise| be [=a new promise=].
+ 1. Let |forwardReaderError| be the following steps, taking a |thisReader| argument:
+  1. [=Upon rejection=] of |thisReader|.[=ReadableStreamGenericReader/[[closedPromise]]=] with reason
+     |r|,
+   1. If |thisReader| is not |reader|, return.
+   1. Perform ! [$ReadableByteStreamControllerError$](|branch1|.[=ReadableStream/[[controller]]=],
+      |r|).
+   1. Perform ! [$ReadableByteStreamControllerError$](|branch2|.[=ReadableStream/[[controller]]=],
+      |r|).
+   1. If |canceled1| is false or |canceled2| is false, [=resolve=] |cancelPromise| with undefined.
+ 1. Let |pullWithDefaultReader| be the following steps:
+  1. If |reader| [=implements=] {{ReadableStreamBYOBReader}},
+   1. Assert: |reader|.[=ReadableStreamBYOBReader/[[readIntoRequests]]=] is [=list/is empty|empty=].
+   1. Perform ! [$ReadableStreamReaderGenericRelease$](|reader|).
+   1. Set |reader| to ! [$AcquireReadableStreamDefaultReader$](|stream|).
+   1. Perform |forwardReaderError|, given |reader|.
+  1. Let |readRequest| be a [=read request=] with the following [=struct/items=]:
+   : [=read request/chunk steps=], given |chunk|
+   ::
+    1. [=Queue a microtask=] to perform the following steps:
+     1. Set |reading| to false.
+     1. Let |chunk1| and |chunk2| be |chunk|.
+     1. If |canceled1| is false and |canceled2| is false, set |chunk2| to ?
```

Comment:
Hmm, good point. In that case, *think* we're going to need to cancel the stream and error both branches?

It looks like this problem already existed in [the current ReadableStreamTee](https://streams.spec.whatwg.org/commit-snapshots/9beefc2ac041a1a848088962d56eb03e89a8bf07/#readable-stream-tee)... We should fix that as well. 🤔

Reply:
Great catch. I think this changed in https://github.com/whatwg/streams/commit/63b4407f424d6ad042852b3d225b7c0ffa116b2d . [Before](https://streams.spec.whatwg.org/commit-snapshots/c63ae9eb7de6069f0e3cf5ea6b0ef68a2c8a3971/#readable-stream-tee), the failure was in a promise handler for a promise that was explicitly marked as [[PromiseIsHandled]]. So I guess such failures were ignored.

These days they go to HTML's "report an exception" for all unhandled exceptions.

Also related: I think the `? ReadableStreamDefaultControllerEnqueue` steps inside the chunk steps should be `!`. (Maybe all call sites for it should be??)

Anyway, what is the correct behavior? I think another option would be to only error branch 2. WDYT? @ricea, any ideas on what makes the most sense from the perspective of how implementations use tee-with-cloneForBranch2-set?

Reply:
> Also related: I think the `? ReadableStreamDefaultControllerEnqueue` steps inside the chunk steps should be `!`. (Maybe all call sites for it should be??)

Indeed, it will only throw if `strategy.size()` throws, but that can never happen since we're using the default strategy. This is the same reasoning as we had for `ReadableStream.from()` in [#1083](https://github.com/whatwg/streams/pull/1083#discussion_r520886364).

> Anyway, what is the correct behavior? I think another option would be to only error branch 2. WDYT?

Well, we shouldn't ignore the failure like we did previously, since that means both branches get "stuck" (never receive a new chunk, or become closed or errored).

Only erroring branch 2 seems too risky. One place where we'll want to tee a readable byte stream is when Fetch clones the request to pass to the service worker (step 5.1 of [HTTP fetch](https://fetch.spec.whatwg.org/commit-snapshots/3babdcb8d1e564040b30bc824420ade870709e4f/#concept-http-fetch)). If the service worker reads the request body but does *not* provide a response (i.e. its `fetch` event handler does not call `event.respondWith()`), then we'll skip step 5.5 and we'll continue the fetch with the original request. However, if we've *failed to clone* a chunk, then the service worker would have successfully read the chunk (because its branch was providing the BYOB request), but our original request's body would have become errored, breaking the fetch.

To me, it seems better to error both branches if we can't clone the chunk.